File: packages/zodiac-dart/lib/src/models/zodiac_element.dart
/// Zodiac element model.
class ZodiacElement {
  final String id;
  final String name;
  final String system; // 'western' | 'chinese'
  final List<String> members;

  const ZodiacElement({
    required this.id,
    required this.name,
    required this.system,
    required this.members,
  });
}

File: packages/zodiac-js/src/models/zodiac-element.ts
/**
 * Zodiac element model.
 *
 * Represents an element in either the Western or Chinese Zodiac system.
 */
export interface ZodiacElement {
  /** Stable identifier (e.g., "fire", "water", "wood") */
  readonly id: string;

  /** English display name */
  readonly name: string;

  /** Which zodiac system this element belongs to */
  readonly system: 'western' | 'chinese';

  /** Sign/animal IDs associated with this element */
  readonly members: readonly string[];
}
